code/controllers/subsystem/puddle.dm

```diff
@@ -0,0 +1,23 @@
+SUBSYSTEM_DEF(puddle)
+	name = "Puddle"
+	wait = 5
+	runlevels = RUNLEVEL_GAME
+	var/list/puddlelist
+
+/datum/controller/subsystem/puddle/Initialize(start_timeofday)
+	. = ..()
+	puddlelist = list()
+
+/datum/controller/subsystem/puddle/stat_entry(msg)
+	..("P: [length(puddlelist)]")
+
+/datum/controller/subsystem/puddle/fire(resumed)
+	// used on the machines subsystem, apparently makes this faster
+	var/list/pudl = src.puddlelist
```

Comment:
```suggestion

```

Reply:
This is apparently a cache-like hack.

It's used in the machine subsystem too